cli/core/test/e2e/framework/config_operations.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package framework
+
+// ConfigOps performs "tanzu config" command operations
+type ConfigOps interface {
+	ConfigSetFeature(path, value string) error
+	ConfigFeatureFlagExists(flagName, value string) (bool, error)
+	ConfigFeatureFlagNotExists(flagName string) (bool, error)
+	ConfigUnsetFeature(path string) error
+	ConfigInit() error
+	ConfigServerList() error
+	ConfigServerDelete(serverName string) error
+}
+
+// configOps is the implementation of ConfOps interface
+type configOps struct {
+	CmdOps
+}
+
+func NewConfOps() ConfigOps {
+	return &configOps{
+		CmdOps: NewCmdOps(),
+	}
+}
+
+// ConfigSetFeature sets the tanzu config feature flag
+func (co *configOps) ConfigSetFeature(path, value string) (err error) {
+	confSetComd := ConfigSet + path + " " + value
+	_, _, err = co.Exec(confSetComd)
+	return err
+}
+
+// ConfigFeatureFlagExists validates the existence of feature flag in tanzu config
+func (co *configOps) ConfigFeatureFlagExists(flagName, value string) (bool, error) {
+	getFeatureCmd := ConfigGet
+	featureFlagStr := flagName + ": " + "\"" + value + "\""
+	err := co.ExecContainsString(getFeatureCmd, featureFlagStr)
+	if err != nil {
+		return false, err
+	}
+	return true, err
+}
+
+// ConfigFeatureFlagNotExists validates the existence of feature flag in tanzu config
+func (co *configOps) ConfigFeatureFlagNotExists(flagName string) (bool, error) {
+	getConfCmd := ConfigGet
+	err := co.ExecNotContainsString(getConfCmd, flagName+": ")
+	if err != nil {
+		return false, err
+	}
+	return true, err
+}
```

Comment:
I think instead of implementing functions like `ConfigFeatureFlagNotExists` and `ConfigFeatureFlagExists` here, it would be better to implement function like `ConfigGetFeatureFlag` which parses the yaml and returns the feature-flag value if it exists else returns error.

As part of the tests, we can just to `ConfigGetFeatureFlag` to get the feature-flag value and compare the result as part of the tests.

Reply:
Yes thats better API, i have created the new API `ConfigGetFeatureFlag`